### PR TITLE
feat: Resource to create roles

### PR DESCRIPTION
**Pull request by ANUGRAHG**
## Purpose
Added role-specific resources in the provider for directory, global account, and subaccount.
This PR will resolve issue #841 

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code via automated test

```bash
go test ./...
```

* Additional test steps

```
...
```

## What to Check

Verify that the following are valid:

* Automated tests are executed successfully

* ...

## Checklist for reviewer

The following organizational tasks must be completed before merging this PR:

* [x] The PR is assigned to the Terraform project and a status is set (typically "in review").
* [x] The PR has the matching labels assigned to it.
* [x] The PR has a milestone assigned to it.
* [x] If the PR closes an issue, the issue is referenced.
* [x] Possible follow-up items are created and linked.

### CODE REVIEW

File: internal/provider/resource_directory_role_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestResourceDirectoryRole(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
```
Comment (lechnerc77) [inline internal/provider/resource_directory_role_test.go:13]: The test should also comprise an update and an import to check all available operations.

File: internal/provider/resource_globalaccount_role_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestResourceGlobalAccountRole(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
+		rec, user := setupVCR(t, "fixtures/resource_globalaccount_role")
+		defer stopQuietly(rec)
+
+		resource.Test(t, resource.TestCase{
```
Comment (lechnerc77) [inline internal/provider/resource_globalaccount_role_test.go:17]: The test should also comprise an update and an import to check all available operations.

File: internal/provider/resource_subaccount_role_test.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestResourceSubAccountRole(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
+		rec, user := setupVCR(t, "fixtures/resource_subaccount_role")
+		defer stopQuietly(rec)
+
+		resource.Test(t, resource.TestCase{
```
Comment (lechnerc77) [inline internal/provider/resource_subaccount_role_test.go:17]: The test should also comprise an update and an import to check all available operations.

**Review by lechnerc77 — CHANGES_REQUESTED** [review]
The tests for the `UPDATE` and `IMPORT` operations should be added to the test cases to ensure that we cover all available operations in the tests.

In addition, I think the test cases need to be re-recorded as they currently run into errors.

File: internal/provider/resource_globalaccount_role_test.go (hunk 2 above)
Comment (ANUGRAHG) [reply internal/provider/resource_globalaccount_role_test.go:17]: Added tests for import and update

File: internal/provider/resource_directory_role_test.go (hunk 1 above)
Comment (ANUGRAHG) [reply internal/provider/resource_directory_role_test.go:13]: Done

**Review by vipinvkmenon — APPROVED** [review]

**Review by lechnerc77 — APPROVED** [review]
lgtm

**Pull request merged**